### PR TITLE
Infix to Postfix Converter in JavaScript 

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#5082 )

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Sir please review it @ankitaggarwal23 @gat786 @siddharth25pandey @atarax665

File: JavaScript/README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@
 - [Recursive preorder traversal in binary search tree](./ds/Recursive_preorder.js)
 - [Recursive inorder traversal in binary search tree](./ds/Recursive_inorder.js)
 - [Recursive postorder traversal of binary search tree](./ds/Recursive_postorder.js)
+- [Infix to Postfix Conversion in JavaScript](./ds/infixtopostfix.js)
```
Comment (ankitaggarwal23) [inline JavaScript/README.md]: Rewrite it as ` [Infix to Postfix Conversion]`

File: JavaScript/ds/infixtopostfix.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+// function that convert infix to postfix
+function InfixtoPostfix(infixval)
+{
+    var postfix=[];
+    var temp=0;
+    push('@');
+   
+
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js:71]: Remove extra one line

File: JavaScript/ds/infixtopostfix.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            }
+            else if(el=='(')
+            {
+                push(el);
+            }
+  // compare precedency 
+            else if(precedency(el)>precedency(stackarr[topp]))
+            {
+                push(el);
+            }
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js]: Align your code with proper format and remove redundant comments

File: JavaScript/ds/infixtopostfix.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    console.log("Postfix Expression:- "+st);
+ }
+ 
+//  Caling Function InfixtoPostfix
+
+ InfixtoPostfix("((a+b)*c)");
+
+
+
+
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js]: Remove extra lines

File: JavaScript/ds/infixtopostfix.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+Output:
+ABC*EF-/+
+
+
+Input:
+a+b*(c^d-e)^(f+g*h)-i
+
+Output:
+abcd^e-fgh*+^*+i-
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js]: - Write outputs with proper display message
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Add your code to repl.it and share the link as well.

**Comment by shivammahajancse** [comment]
Repl link:- https://replit.com/join/hoabtdsf-shivammahajan4
All requested changes done @ankitaggarwal23

File: JavaScript/ds/infixtopostfix.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        var el=infixval[i];
+        if(operator(el))
+        {
+            if (el ==')') {
+                while (stackarr[topp] != "(") {
+                  postfix[temp++] = pop();
+                }
+          pop();
+            }
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js]: Give a tab space before `pop()`

File: JavaScript/ds/infixtopostfix.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+ }
+ 
+//  Caling Function InfixtoPostfix
+
+ InfixtoPostfix("((a+b)*c)");
+
+ /*
+OutPut:
+ab+c*
+
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js]: Remove this lines

File: JavaScript/ds/infixtopostfix.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+A+B*C/(E-F)
+
+Output:
+Postfix Expression: ABC*EF-/+
+
+Input:
+a+b*(c^d-e)^(f+g*h)-i
+
+Output:
+Postfix Expression: abcd^e-fgh*+^*+i-
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtopostfix.js]: Write proper display message while reading and printing the outputs

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
Please review @ankitaggarwal23 sir

File: JavaScript/ds/infixtopostfix.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    }    
+
+    // Print postfix expression in console
+    
+    console.log("Postfix Expression:- "+st);
+ }
+ 
+//  Caling Function InfixtoPostfix
+
+ InfixtoPostfix("((a+b)*c)");
```
Comment (atarax665) [inline JavaScript/ds/infixtopostfix.js]: Modify it for dynamic inputs

File: JavaScript/ds/infixtopostfix.js [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+                {
+                     postfix[temp++]=pop();
+                }
+                push(el);
+            }
+        }
+        else{
+            postfix[temp++]=el;
+        }
+
```
Comment (atarax665) [inline JavaScript/ds/infixtopostfix.js]: Remove the extra blank line.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Requested some minor fixes before we merge this PR.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**